src/lib/verifier/prod.ml

```diff
@@ -44,9 +44,24 @@ module Worker_state = struct
           (let bc_vk = Precomputed_values.blockchain_verification ()
            and tx_vk = Precomputed_values.transaction_verification () in
            let module M = struct
-             let verify_commands (_cs : User_command.Verifiable.t list) :
-                 _ list =
-               failwith "unimplemented"
+             let verify_commands (cs : User_command.Verifiable.t list) :
+                 [ `Valid of Coda_base.User_command.Valid.t
+                 | `Invalid
+                 | `Valid_assuming of
+                   ( Pickles.Side_loaded.Verification_key.t
+                   * Coda_base.Snapp_statement.t
+                   * Pickles.Side_loaded.Proof.t )
+                   list ]
+                 list =
+               List.map cs ~f:(function
+                 | Signed_command c -> (
+                   match Signed_command.check c with
+                   | Some c ->
+                       `Valid (User_command.Signed_command c)
+                   | None ->
+                       `Invalid )
+                 | Snapp_command _ ->
```

Comment:
Do we not have to check snapp commands like we do for proof level `Check| None` below?

Reply:
Fixed. Full verification happens in this PR https://github.com/CodaProtocol/coda/pull/5896/files#diff-a234e91f391560e6e5b65833bffb041cR47